File: github_cpp/10/166.cpp
#include <algorithm>
#include <string>
#include <iostream>
#include "EditDistance.h"

EditDistance::EditDistance(std::string a, std::string b) {
	this->a = a;
	this->b = b;
}

int EditDistance::editDistance(int i, int j) {
	if (i >= a.size() || j >= b.size()) {
		if (i >= a.size())
			return b.size() - j;
		return a.size() - i;
	}

	if (a[i] == b[j])
		return editDistance(i + 1, j + 1);

	return std::min(editDistance(i + 1, j + 1), std::min(editDistance(i + 1, j), editDistance(i, j + 1))) + 1;
}

int EditDistance::editDistance(std::string a, std::string b) {
	EditDistance * dist = new EditDistance(a, b);
	return dist->editDistance(0, 0);
}

File: github_code_index/java/8/166.java
public static void main (String args[]) {
	String a = "apple";
	String b = "anne";
	editdistance(a,b);

}
public static void editdistance(String a, String b){
	int[][] dp = new int[a.length()+1][b.length()+1];
	for (int i = 0; i <= a.length(); i++) {
		dp[i][0] = i;
	}
	for(int j = 0; j <= b.length(); j++) {
		dp[0][j] = j;
	}
	for(int i = 1; i <= a.length(); i++){
		for(int j = 1; j <= b.length(); j++){
			if(a.charAt(i-1) == b.charAt(j-1)){
				dp[i][j] = dp[i-1][j-1];
			}else{
				dp[i][j] = Math.min(dp[i-1][j-1],dp[i-1][j],dp[i][j-1])+1;
			}
		}
	}

	return dp[a.length()][b.length()];

}